casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -128,17 +121,14 @@ object Estimator {
             case (acc, children) =>
               children.filter(scoreMap.contains).toList.foldLeftM(acc) {
                 case (acc2, childHash) =>
-                  for {
-                    blockMetadataOpt <- blockDag.lookup(childHash)
-                    result = blockMetadataOpt match {
-                      case Some(blockMetaData)
-                          if blockMetaData.parents.size > 1 && blockMetaData.sender != validator =>
-                        val currScore       = acc2.getOrElse(childHash, 0L)
-                        val validatorWeight = blockMetaData.weightMap.getOrElse(validator, 0L)
-                        acc2.updated(childHash, currScore + validatorWeight)
-                      case _ => acc2
-                    }
-                  } yield result
+                  blockDag.lookup(childHash).map {
+                    case Some(childBlock)
+                        if childBlock.parents.size > 1 && childBlock.sender != validator =>
+                      val currScore       = acc2.getOrElse(childHash, 0L)
+                      val validatorWeight = childBlock.weightMap.getOrElse(validator, 0L)
+                      acc2.updated(childHash, currScore + validatorWeight)
+                    case _ => acc2
+                  }
```

Comment:
My head hurts when I try to understand this function. Please make it more readable.

Reply:
The change is NFC. Hopefully the after version is better than the before. If not I'll just revert this entirely.

Reply:
I've also just considered removing this part. It was initially an idea to make the estimator a bit more fair but for the amount of thinking required verify its correctness I don't think it is worth it.

Reply:
On further thought, I'm going to remove this indeed.